thirdparty/nginx/nginx-1.7.1-zimbra/src/event/ngx_event_openssl.c

```diff
@@ -896,8 +895,7 @@ ngx_ssl_handshake(ngx_connection_t *c)
         if (ngx_handle_write_event(c->write, 0) != NGX_OK) {
             return NGX_ERROR;
         }
-
-        return NGX_AGAIN;
+        return NGX_ERROR;
```

Comment:
If this change is good then don't need the if statement above - we're always going to return NGX_ERROR?
Doesn't feel right to me but am not sure I understand code enough to say why.  Perhaps needs a comment explaining why?

Reply:
I do agree this is a good question. Does NGX_ERROR imply try again in this case?

Reply:
Yes @tmclane @fatkudu